bin/common.c

```diff
@@ -376,6 +376,17 @@ int s2n_set_common_server_config(int max_early_data, struct s2n_config *config,
 
 int s2n_setup_server_connection(struct s2n_connection *conn, int fd, struct s2n_config *config, struct conn_settings settings)
 {
+    if (settings.deserialize_in) {
+        size_t deserialize_length = 0;
+        GUARD_RETURN(get_file_size(settings.deserialize_in, &deserialize_length), "Failed to read serialize-in file size");
+        ENSURE_RETURN(deserialize_length <= UINT32_MAX, "serialize-in file size is too large");
+        uint8_t *mem = malloc(deserialize_length);
+        GUARD_RETURN_NULL(mem);
+        GUARD_RETURN(load_file_to_array(settings.deserialize_in, mem, deserialize_length), "Failed to read serialize-in file");
+        GUARD_RETURN(s2n_connection_deserialize(conn, mem, (uint32_t) deserialize_length), "Failed to deserialize connection");
```

Comment:
It doesn't look like these errors are actually being caught anywhere:
https://github.com/aws/s2n-tls/blob/3d125a514dfe0a9e4d620f22eb40703088ac53fb/bin/s2nd.c#L224

Should we maybe GUARD_EXIT `s2n_setup_server_connection`?